File: python/examples/info.py
# -*- coding: utf-8 -*-

'''
Example code to call Rosette API to get information such as version and build
'''

import argparse
import pprint
import sys

# enable imports from rosette.api
sys.path += '../../'

from rosette.api import API, Operator

parser = argparse.ArgumentParser(description='Accept Rosette API key')
parser.add_argument('--key', required=True, help='Rosette API key')
parser.add_argument('--service_url', nargs='?', help='Optional user service URL')
args = parser.parse_args()

# Create an API instance
if args.service_url:
    api = API(service_url=args.service_url, user_key=args.key)
else:
    api = API(user_key=args.key)

op = Operator(api, None)
result = op.info()

pprint.pprint(result)

File: nodejs/examples/info.js
/*
 * Example code to call Rosette API to get information such as version and build.
 */

"use strict";

var Api = require("rosette-api").Api;
var ArgumentParser = require("argparse").ArgumentParser;

var parser = new ArgumentParser({
  addHelp: true,
  description: "Get information about Rosette API"
});
parser.addArgument(["--key"], {help: "Rosette API key", required: true});
var args = parser.parseArgs();

var api = new Api(args.key);
api.info(function(err, res) {
  if (err) {
    throw err;
  }
  else {
    console.log(res);
  }
});
